src/Symfony/Component/Console/Tests/Command/InvokableCommandTest.php

```diff
@@ -312,13 +347,29 @@ function (#[Option] ?bool $a = true) {},
             function (#[Option] ?bool $a = false) {},
             'The option parameter "$a" must not be nullable when it has a default boolean value.',
         ];
-        yield 'nullable-string' => [
-            function (#[Option] ?string $a = null) {},
-            'The option parameter "$a" must not have a default of null.',
+        yield 'invalid-union-type' => [
+            function (#[Option] array|bool $a = false) {},
+            'The union type for parameter "$a" is not supported as a command option. Only "bool|string", "bool|int", "bool|float" types are allowed.',
+        ];
+        yield 'union-type-cannot-allow-null' => [
+            function (#[Option] string|bool $a = null) {},
```

Comment:
that would be a php deprecation, right?
```suggestion
            function (#[Option] string|bool|null $a = null) {},
```